src/test/rpc_wallet_tests.cpp

```diff
@@ -439,6 +478,544 @@ BOOST_AUTO_TEST_CASE(rpc_wallet_z_importexport)
+    std::string zaddr1 = pa.ToString();
+    
+    // there are no utxos to spend
+    {
+        std::vector<SendManyRecipient> recipients = { SendManyRecipient(zaddr1,100.0, "DEADBEEF") };
+        std::shared_ptr<AsyncRPCOperation> operation( new AsyncRPCOperation_sendmany(taddr1, {}, recipients, 1) );
+        operation->main();
+        BOOST_CHECK(operation->isFailed());
+        std::string msg = operation->getErrorMessage();
+        BOOST_CHECK( msg.find("Insufficient funds, no UTXOs found") != string::npos);
+    }
+    
+    // there are no unspent notes to spend
+    {
+        std::vector<SendManyRecipient> recipients = { SendManyRecipient(taddr1,100.0, "DEADBEEF") };
+        std::shared_ptr<AsyncRPCOperation> operation( new AsyncRPCOperation_sendmany(zaddr1, recipients, {}, 1) );
+        operation->main();
+        BOOST_CHECK(operation->isFailed());
+        std::string msg = operation->getErrorMessage();
+        BOOST_CHECK( msg.find("Insufficient funds, no unspent notes") != string::npos);
+    }
+
+    // get_memo_from_hex_string())
+    {
+        std::vector<SendManyRecipient> recipients = { SendManyRecipient(zaddr1,100.0, "DEADBEEF") };
+        std::shared_ptr<AsyncRPCOperation> operation( new AsyncRPCOperation_sendmany(zaddr1, recipients, {}, 1) );
+        std::shared_ptr<AsyncRPCOperation_sendmany> ptr = std::dynamic_pointer_cast<AsyncRPCOperation_sendmany> (operation);
+        TEST_FRIEND_AsyncRPCOperation_sendmany proxy(ptr);
         
+        std::string memo = "DEADBEEF";
+        boost::array<unsigned char, ZC_MEMO_SIZE> array = proxy.get_memo_from_hex_string(memo);
+        BOOST_CHECK_EQUAL(array[0], 0xDE);
+        BOOST_CHECK_EQUAL(array[1], 0xAD);
+        BOOST_CHECK_EQUAL(array[2], 0xBE);
+        BOOST_CHECK_EQUAL(array[3], 0xEF);
+        for (int i=4; i<ZC_MEMO_SIZE; i++) {
+            BOOST_CHECK_EQUAL(array[i], 0x00);  // zero padding
+        }
+        
+        // memo is longer than allowed
+        std::vector<char> v (2 * (ZC_MEMO_SIZE+1));
+        std::fill(v.begin(),v.end(), 'A');
+        std::string bigmemo(v.begin(), v.end());
+        
+        try {
+            proxy.get_memo_from_hex_string(bigmemo);
+        } catch (const Object& objError) {
+            BOOST_CHECK( find_value(objError, "message").get_str().find("too big")!= string::npos);
+        }
+        
+        // invalid hexadecimal string
+        std::fill(v.begin(),v.end(), '@'); // not a hex character
+        std::string badmemo(v.begin(), v.end());
+        
+        try {
+            proxy.get_memo_from_hex_string(badmemo);
+        } catch (const Object& objError) {
+            BOOST_CHECK( find_value(objError, "message").get_str().find("hexadecimal format")!= string::npos);
+        }
+        
+        // odd length hexadecimal string
+        std::fill(v.begin(),v.end(), 'A');
+        v.resize(v.size() - 1);
+        BOOST_CHECK(v.size() %2 == 1); // odd length
```

Comment:
Nit: This is an assertion of consistency within the test, so use ``assert`` rather than ``BOOST_CHECK``.